oci/table_oci_core_nat_gateway.go

```diff
@@ -0,0 +1,242 @@
+package oci
+
+import (
+	"context"
+	"strings"
+
+	oci_common "github.com/oracle/oci-go-sdk/v36/common"
+	"github.com/oracle/oci-go-sdk/v36/core"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableCoreNatGateway(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_core_nat_gateway",
+		Description: "OCI Core Nat Gateway",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AnyColumn([]string{"id"}),
+			Hydrate:    getCoreNatGateway,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listCoreNatGateways,
+		},
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "id",
+				Description: "The OCID of the NAT gateway.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "display_name",
+				Description: "A user-friendly name of the NAT gateway.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the NAT gateway was created",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The NAT gateway's current state.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "vcn_id",
+				Description: "The OCID of the VCN the NAT gateway belongs to.",
```

Comment:
can you please put `lifecycle_state` and `vcn_id` above `time_created` as these two columns are more important.